bpf/include/bpf/csum.h

```diff
@@ -30,13 +30,17 @@ static __always_inline __wsum csum_diff(const void *from, __u32 size_from,
 					__u32 seed)
 {
 	if (__builtin_constant_p(size_from) &&
-	    __builtin_constant_p(size_to) &&
-	    __builtin_constant_p(seed) && seed == 0) {
+	    __builtin_constant_p(size_to)) {
 		/* Optimizations for frequent hot-path cases that are tiny to just
 		 * inline into the code instead of calling more expensive helper.
 		 */
-		if (size_from == 4 && size_to == 4)
+		if (size_from == 4 && size_to == 4 &&
+		    __builtin_constant_p(seed) && seed == 0)
```

Comment:
The basic idea was to resolve at compilation time the fact that I can bypass calling csum_diff_external() entirely and fully use one of the inlined version instead. Hence the tests on __builtin_constant_p() first. If we'd ever have a case where size_* is not constant then I wanted to avoid bloating/increasing verifier complexity of the code (esp. for 4.9 kernels).